synapse/lib/ast.py

```diff
@@ -511,6 +510,7 @@ async def todogenr():
                 if doedges:
                     async for verb, n2iden in node.iterEdgesN1():
                         edges.append((n2iden, {'verb': verb}))
+                        await asyncio.sleep(0)
```

Comment:
This is definitely a hot loop; but should the Node.iterEdgesN1() and Node.iterEdgesN2 APIs yield instead?

Reply:
Might be an ever better optimization to mergegenr the edges from the layers rather than using a spooled set...

Reply:
( i'd like to keep this sleep(0) higher in the call stack so migrations etc can zoom